jmespath-core/src/main/java/io/burt/jmespath/Adapter.java

```diff
@@ -32,6 +33,12 @@
   String toString(T value);
 
   /**
+   * Converts the argument to a Double, or null if the argument does not
+   * represent a number.
+   */
+  Double toDouble(T value);
```

Comment:
Why `Double` (and not `double`)? Is it so that you could avoid a call to `isNumber` and get `null` in that case (instead of raising an exception)? If you are going for an object, why not use the base-class `Number` instead?